src/Umbraco.Web/PropertyEditors/ValueConverters/ContentPickerPropertyConverter.cs

```diff
@@ -108,11 +110,29 @@ public override object ConvertSourceToObject(PublishedPropertyType propertyType,
             {
                 if ((propertyType.PropertyTypeAlias != null && PropertiesToExclude.Contains(propertyType.PropertyTypeAlias.ToLower(CultureInfo.InvariantCulture))) == false)
                 {
-                    var content = UmbracoContext.Current.ContentCache.GetById((int)source);
-                    return content;
+                    IPublishedContent content;
+                    switch (propertyType.PropertyEditorAlias)
+                    {
+                        case Constants.PropertyEditors.ContentPickerAlias:
+                            int sourceInt;
+                            if (int.TryParse(source.ToString(), out sourceInt))
+                            {
+                                content = UmbracoContext.Current.ContentCache.GetById(sourceInt);
+                                return content;
+                            }
+                            break;
+                        case Constants.PropertyEditors.ContentPicker2Alias:
+                            Udi udi;
+                            if (Udi.TryParse(source.ToString(), out udi))
```

Comment:
if `source` is not null, it should be a `Udi` already (see `ConvertDataToSource`) so probably you can try to cast (`source as Udi`) instead of re-parsing?